File: PersonalProjects/prosjekt-3/data/scraper/writer.py

from uuid import uuid4

class Writer:
    writers = {}
    def __init__(self, writer):
        self.id = uuid4()
        self.writer = writer
        Writer.writers[writer] = self.id

    def getWriter(self, name):
        if name in Writer.writers.keys():
            return Writer.writers[name]
        else:
            return Writer(name).id

File: PersonalProjects/prosjekt-3/backend/crud/writer.ts
import { Client, QueryResultRow } from "pg";
import { Writer } from "./../types";

export const getWriters = async (movie: QueryResultRow, client: Client) => {
  const result = await client.query(
    "select w.name, w.id, mw.role from moviewriter mw inner join writer w on w.id = mw.writerid where mw.movieid = $1::uuid",
    [movie.id]
  );
  return result.rows;
};
